staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -71,20 +71,30 @@ type EgressSelection struct {
 
 // Connection provides the configuration for a single egress selection client.
 type Connection struct {
-	// Type is the type of connection used to connect from client to konnectivity server.
-	// Currently supported values are "http-connect" and "direct".
-	Type string
+	// Protocol is the protocol used to connect from client to the konnectivity server.
+	// Supported values are "http-connect," "grpc", and "direct"
+	Protocol string
 
-	// httpConnect is the config needed to use http-connect to the konnectivity server.
+	// Transport is the transport socket used to connect to konnectivity server.
+	// Supported values are "uds" and "tcp". TCP will only work with http-connect protocol
 	// +optional
-	HTTPConnect *HTTPConnectConfig
-}
+	Transport string
```

Comment:
Following the api convention on [unions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#unions), how about
```golang
type TCPTransport struct {
  URL string
  TLSConfig *TLSConfig
}

type UDSTransport struct {
  UDSName string
}

type Connection struct {
  Protocol ProtocolType
  TCP *TCPTranport
  UDS *UDSTransport
}
```